test/src/sdk-harness/test_projects/call_frames/mod.rs

```diff
@@ -41,14 +44,62 @@ async fn can_get_code_size() {
 async fn can_get_first_param() {
     let (instance, _id) = get_call_frames_instance().await;
     let result = instance.get_first_param().call().await.unwrap();
-    assert!(is_within_range(result.value));
+    // Hash the function name with Sha256
+    let mut hasher = Sha256::new();
+    let function_name = "get_first_param()";
+    hasher.update(function_name);
+    let function_name_hash = hasher.finalize();
+    // Grab the first 4 bytes of the hash per https://github.com/FuelLabs/fuel-specs/blob/master/specs/protocol/abi.md#function-selector-encoding
+    let function_name_hash = &function_name_hash[0..4];
+    // Convert the bytes to decimal value
+    let selector = function_name_hash[3] as u64 + 256 * (function_name_hash[2] as u64 + 256 * (function_name_hash[1] as u64 + 256 * function_name_hash[0] as u64));
+    assert_eq!(result.value, selector);
 }
 
 #[tokio::test]
-async fn can_get_second_param() {
+async fn can_get_second_param_u64() {
```

Comment:
It's good to see that there is a mixture of params unfortunately this is not an exhaustive list. For these tests to thoroughly prove that the second param works as expected we would need to perform at least a check on each native type and it only gets more complex considering the permutations.

That being said having 1 test for each native type and then another few for a mixture of types would generally suffice as a bug being discovered in the permutations is an unlikely edge case if each native type works and a few tests which mix also pass.

Could you add some tests to make sure that each native type is checked?

Reply:
@mohammadfawaz what do you think?

Reply:
hm, I see. As long as `read()` is thoroughly tested then this does not need to be

Reply:
I could add a test for a mixture of reference and copy types as [this](https://github.com/FuelLabs/fuels-rs/issues/470) was just merged.

Reply:
This looks like an SDK issue rather than a Sway issue so I don't believe that to be necessary